src/components/settings/SettingsHeader/index.tsx

```diff
@@ -9,21 +12,38 @@ import { AppRoutes } from '@/config/routes'
 import { useHasFeature } from '@/hooks/useChains'
 import { FEATURES } from '@/utils/chains'
 
+type NavItem = {
+  href: string
+  label: string
+}
+
+type HideConditions = Record<string, boolean>
+
+const filterRoutes = (navItems: NavItem[], hideConditions: HideConditions) => {
+  return navItems.filter((item) => !Boolean(hideConditions[item.href]))
```

Comment:
Doesn't matter much, but you don't need the `Boolean` here, `!` already coerces it to one.